### ISSUE TITLE
Design | Discovery | Profile | Screenreader Accessibility | Nametag 

### ISSUE THREAD

**Issue by ericasauv09** [department-of-veterans-affairs/va.gov-team#47266]
## Background

The Accessibility team informed us that a screenreader may not pick up the Nametag section of Profile because it is not under an H2. Currently, the Nametag is hidden under the first heading so by putting it under its own H2, under the first H1, screenreaders will be able to identify it better. 

Accessibility had some recommendations for consideration as we evaluate options including: 

- Angela considers this to be "basic information," which could be displayed right below the H1 under a new H2 heading that says “basic information”. This will make it visible to the screen reader.
- If we put it _above_ H1, we might not need to display it on each page, as it could get redundant. Instead, the Nametag could be included and read out on Personal information page only.
- Or for less priority, it could be at the bottom of the page

## Tasks
- [ ] Consider alternatives for Nametag info 
- [ ]  Review viability and ease of implementation with FE 
- [ ] Create 2 - 3 mock-ups
- [ ] Review with team (and update mocks if needed)
- [ ] Review with MY VA (and update mocks if needed)

## Acceptance Criteria
- [ ] Align on design

**Comment by SKasimow** [comment]
Hi @cgednalske - I updated this ticket to reflect Design/Discovery work and kept it focused on looking at alternatives for Nametag placement and aligning team.  Let me know if that sounds good or if there is anything else you'd like to add/change. Thanks!

**Comment by andaleliz** [comment]
@cgednalske I estimated this as a 5 for planning purposes, but feel free to update that when you're back if you feel it's over/under estimated.

**Comment by cgednalske** [comment]
Hi @Samara-Strauss and @SKasimow -- a few updates on this work. 

To get a better sense of the nametag's progression and the basis for the information included, I dug into [past tickets](https://github.com/department-of-veterans-affairs/va.gov-team/issues/13918) and designs for the nametag, as well as Angela A's [recent work](https://github.com/department-of-veterans-affairs/va.gov-team/issues/46714) auditing the design logic. 

With the primary purpose being to visually unite the Veteran's authenticated space and clearly present their disability rating, the name tag is definitely doing this and doesn't seem to require any visual changes. As far as I can tell, we haven't heard any negative feedback from sighted users and, in at least two research studies, Liz and I have actually heard that users liked seeing the disability rating front and center. But of course we also want the experience to be a good one for screen reader users and anyone else who might miss the nametag. 

With this in mind, I met with Tiffany and Angela Fowler today to discuss which of their earlier proposed solutions might work best. We ended up agreeing on the one with fewer changes to the design. Here are their suggested steps: 

(_Note: the suggested approach combines all of these steps_)

### 1. **Visually leave nametag as is** 
- There were initially some other suggestions from Angela that involved moving the nametag. Now, with more background on the visual goals, we all agreed it doesn't make sense to move it elsewhere. As is works well to visually unite My VA and Profile. 

### 2. **Make some code changes** 
- To make the experience better for screen reader users, we can make some code changes so the name tag info becomes easier to find.
- We can put in a **landmark role** such as banner role to better identify different regions of the page (this is a screen reader specific place marker that indicates different regions of a page). @adamwhitlock1, Tiffany sent along [this resource article](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/landmark_role) if helpful.
- They do NOT suggest using aria live at all to announce this.

### 3. **Add full name and disability rating to main Profile info** 
- We discussed how this information is important enough that including full name and disability rating in the main sections of Profile would be beneficial for screen reader users or anyone else who might miss this info in the name tag. 
- They also suggested adding disability rating to My VA, but I don't believe My VA is really the place for this type of information (but please correct me if I'm wrong on this).
- **_Note: we may have additional user feedback supporting this._** I'll have to check, but I think I recall there being at least one instance in Personal info research where a Veteran had trouble finding their full name in the nametag and was looking for it on Personal info.

Would love to hear your thoughts on this approach!

-----

**As a separate note:** This exploration also ran into another related but larger question about the relationship between My VA and Profile overall: whether the nametag on its own does enough to unite the two or if there might be more that could be explored with the navigation/connection. That is definitely a larger question that expands beyond the nametag and seems to have a pretty detailed history, but wanted to make sure it is documented in case helpful.

**Comment by Samara-Strauss** [comment]
All of this sounds great to me!

I do have a question about the code changes. I want to make sure that the nametag doesn't get read out on each page of the profile, because I don't think it needs to, and my hypothesis is that it would be annoying and redundant for screen reader users. Can you clarify how and when the nametag would get read out? I trust that you all talked about this, it's just not totally clear to me. We can meet to chat if some of these details are easier to talk about than type up.

**Comment by SKasimow** [comment]
Hi @cgednalske - I like the idea of making the Nametag easier for screenreaders to find by using **landmark role** (if Adam doesn't have any concerns).

I can see how adding the full name and disability rating to the Profile itself (assume on Personal info page) would be helpful but not sure how this would look visually on the page when shown 2x. A mock would help if we decide to pursue this option.

**Comment by cgednalske** [comment]
@Samara-Strauss great question and I had the same 😄 This wouldn't actually "read out" the alert at all (as far as forcefully announcing it). It would just do some restructuring to make it much easier for screen reader users to be aware that the nametag information is there - should they chose to have it read out. I was also worried about this reading the same thing out a bunch of times, but they assured me this would not do that and simply reorganize the information in a more helpful manner.

@SKasimow also a good consideration! Their reasoning here is that if it's important enough that sighted users would need/want to see it right away, then screen reader users should be able to access it within the main information - with access to info coming first as long as visuals aren't problematic. Definitely you're right that how it looks visually would depend on how full name and disability are added. What's promising is that we already included branch of service in main info along with the nametag and it's never been strange to have that in two places (as far as I know). I'd imagine the additions would be minimal, but also would be interested to see mockups.

As far as where the info would go, I have assumptions, but also questions, particularly around disability rating and how important it is (does it warrant it's own section?) /how much info Veterans would want to see (is there connected info we don't show right now but might want to?). Because this is IA-related, it also might be something that would benefit from a card sorting study.

**Comment by Samara-Strauss** [comment]
@cgednalske thank you for clarifying about how the screen reader code would handle the nametag! This clears up any confusion I had.

FWIW, I do think the right call is to pull full name and disability rating into the profile somehow. I'm not sure exactly how and it might take some iteration to get things right, but I do think it's the right call.

**Comment by SKasimow** [comment]
Thanks for the adt'l info; that's helpful @cgednalske

**Comment by cgednalske** [comment]
@adamwhitlock1 FYI on nametag changes 😃

**Comment by SKasimow** [comment]
**"Landmark role!!"**  That's the term that escaped me on today's stand-up. 

Adam sound like ok approach?

**Comment by cgednalske** [comment]
@SKasimow closing this out as we discussed in a recent standup 👍
